lib/resources/etc_libuser.rb

```diff
@@ -0,0 +1,108 @@
+require 'utils/simpleconfig'
+require 'utils/object_traversal'
+
+module Inspec::Resources
+  class EtcLibUser < Inspec.resource(1)
+    name 'etc_libuser'
+    desc 'Use the etc_libuser InSpec audit resource to check the configuration of /etc/libuser.conf.
+          Due to limitations of the RSpec engine and for consistency, please use the method described here
+          (https://github.com/chef/inspec/issues/1543) for accessing the hierarchical values in
+          /etc/libuser.conf'
+    example "
+          ---start sample /etc/libuser.conf---
+          # See libuser.conf(5) for more information.
+          ...
+          [import]
+          # Data from these files is used when libuser.conf does not define a value.
+          # The mapping is documented in the man page.
+          login_defs = /etc/login.defs
+          default_useradd = /etc/default/useradd
+          ...
+          ---end sample /etc/libuser.conf---
+
+          describe etc_libuser do
+            its(['import','login_defs']){should eq ['/etc/login.defs']}
+          end
+         "
+
+    include ObjectTraverser
+    attr_reader :params, :raw_content
+
+    class UnknownEtcLibUserConf < StandardError; end
+
+    def initialize(conf_path = nil)
+      @conf_path = conf_path || inspec.os_env('LIBUSER_CONF').content || '/etc/libuser.conf'
+      file = inspec.file(@conf_path)
+      return skip_resource "The `etc_libuser` resource is not supported on your OS. Can't find '#{@conf_path}'." unless file.file?
```

Comment:
I think this should actually be a failure. If someone writes a test and the file isn't there, we cannot perform the requested assertion. We've recently changed to use exceptions for this also so it makes it easier to raise them anywhere in a resource, not just the `initialize` method.

I'd change this to:

```ruby
raise Inspec::Exceptions::FailResource, "Can't find '#{@conf_path}'" unless file.file?
```